src/test/java/seedu/address/logic/commands/AddStudentCommandTest.java

```diff
@@ -140,17 +143,29 @@ public ReadOnlyAddressBook getAddressBook() {
 
         @Override
         public boolean hasPerson(Person person) {
-            throw new AssertionError("This method should not be called.");
+            requireNonNull(person);
+            return persons.stream().anyMatch(p -> p.isSamePerson(person));
         }
 
         @Override
         public void deletePerson(Person target) {
-            throw new AssertionError("This method should not be called.");
+            requireNonNull(target);
+            persons.remove(target);
+            markedAttendance.remove(target);
         }
 
         @Override
         public void setPerson(Person target, Person editedPerson) {
-            throw new AssertionError("This method should not be called.");
+            requireAllNonNull(target, editedPerson);
+            int index = persons.indexOf(target);
+            if (index == -1) {
+                throw new AssertionError("This method should not be called.");
```

Comment:
Might want to change the message of the Assertion Error to better explain the issue, like "No such person exists"